.github/workflows/analysis.yml

```diff
@@ -85,7 +118,7 @@ jobs:
 
   merge:
     runs-on: ubuntu-latest
-    needs: build
+    needs: [build, long]
```

Comment:
Unfortunately I don't think this splitting approach will work. The caching seems to have accelerated the startup time down to 1 minute, but the `long` tests aren't being run. I see the runs on your branch and the `long` tests are taking just the 1 minute startup time. From the logs it looks like they're being skipped, but I don't see why.

But I don't think this approach will work because as an open source org, we have 20 (I believe) concurrent runners and that's shared across all workflows so moving jobs around like this won't have any effect.